relay/polkadot/src/weights/runtime_common_identity_migrator.rs

```diff
@@ -0,0 +1,104 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Autogenerated weights for `runtime_common::identity_migrator`
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 32.0.0
+//! DATE: 2024-02-29, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! WORST CASE MAP SIZE: `1000000`
+//! HOSTNAME: `runner-bn-ce5rx-project-674-concurrent-0`, CPU: `Intel(R) Xeon(R) CPU @ 2.60GHz`
+//! WASM-EXECUTION: `Compiled`, CHAIN: `Some("rococo-dev")`, DB CACHE: 1024
+
+// Executed Command:
+// ./target/production/polkadot
+// benchmark
+// pallet
+// --chain=rococo-dev
+// --steps=50
+// --repeat=20
+// --no-storage-info
+// --no-median-slopes
+// --no-min-squares
+// --pallet=runtime_common::identity_migrator
+// --extrinsic=*
+// --execution=wasm
+// --wasm-execution=compiled
+// --header=./polkadot/file_header.txt
+// --output=./polkadot/runtime/rococo/src/weights/runtime_common_identity_migrator.rs
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+#![allow(missing_docs)]
+
+use frame_support::{traits::Get, weights::Weight};
+use core::marker::PhantomData;
+
+/// Weight functions for `runtime_common::identity_migrator`.
+pub struct WeightInfo<T>(PhantomData<T>);
+impl<T: frame_system::Config> polkadot_runtime_common::identity_migrator::WeightInfo for WeightInfo<T> {
+	/// Storage: `Identity::IdentityOf` (r:1 w:1)
+	/// Proof: `Identity::IdentityOf` (`max_values`: None, `max_size`: Some(7572), added: 10047, mode: `MaxEncodedLen`)
+	/// Storage: `Identity::SubsOf` (r:1 w:1)
+	/// Proof: `Identity::SubsOf` (`max_values`: None, `max_size`: Some(3258), added: 5733, mode: `MaxEncodedLen`)
+	/// Storage: `System::Account` (r:2 w:2)
+	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
+	/// Storage: `Dmp::DeliveryFeeFactor` (r:1 w:0)
+	/// Proof: `Dmp::DeliveryFeeFactor` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `XcmPallet::SupportedVersion` (r:1 w:0)
+	/// Proof: `XcmPallet::SupportedVersion` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Dmp::DownwardMessageQueues` (r:1 w:1)
+	/// Proof: `Dmp::DownwardMessageQueues` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Dmp::DownwardMessageQueueHeads` (r:1 w:1)
+	/// Proof: `Dmp::DownwardMessageQueueHeads` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Identity::SuperOf` (r:0 w:100)
+	/// Proof: `Identity::SuperOf` (`max_values`: None, `max_size`: Some(114), added: 2589, mode: `MaxEncodedLen`)
+	/// The range of component `r` is `[0, 20]`.
+	/// The range of component `s` is `[0, 100]`.
+	fn reap_identity(r: u32, s: u32, ) -> Weight {
```

Comment:
@seadanda done here: https://github.com/joepetrowski/runtimes/pull/3 for Kusama, people-polkadot, people-kusama

Reply:
@joepetrowski @seadanda but polkadot's benchmark does not work:
```
Running benchmarks for runtime_common::identity_migrator to ./polkadot-weights/
2024-05-30 20:23:30 Starting benchmark: runtime_common::identity_migrator::reap_identity    
2024-05-30 20:23:35 Running  benchmark: runtime_common::identity_migrator.reap_identity(2 args) 32/100 1/1    
2024-05-30 20:23:40 can_check_out(destination: Location { parents: 0, interior: X1([Parachain(1004)]) }, asset: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(4687100000) }, _) error: NotDepositable    
Error: 
   0: Invalid input: Benchmark runtime_common::identity_migrator::reap_identity failed: CannotCheckOutTeleport
```

I just ran it with `RUST_LOG=runtime::on_reap_identity=trace,xcm::fungible_adapter=trace`, and it tries to can_check_out the amount `205217100000`, but when it fails the amount is `4687100000` all the time.

```...
2024-05-30 21:13:44.699 TRACE main xcm::fungible_adapter: can_check_out dest: Location { parents: 0, interior: X1([Parachain(1004)]) }, what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(205217100000) }    
2024-05-30 21:13:44.699 TRACE main xcm::fungible_adapter: check_out dest: Location { parents: 0, interior: X1([Parachain(1004)]) }, what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(205217100000) }    
2024-05-30 21:13:44.712 TRACE main xcm::fungible_adapter: withdraw_asset what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(205217100000) }, who: Location { parents: 0, interior: X1([AccountId32 { network: None, id: [35, 154, 38, 60, 227, 140, 132, 230, 230, 19, 67, 42, 230, 243, 159, 250, 54, 60, 237, 56, 213, 43, 32, 204, 18, 211, 35, 95, 86, 125, 200, 237] }]) }    
2024-05-30 21:13:44.712 TRACE main xcm::fungible_adapter: can_check_out dest: Location { parents: 0, interior: X1([Parachain(1004)]) }, what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(205217100000) }    
2024-05-30 21:13:44.712 TRACE main xcm::fungible_adapter: check_out dest: Location { parents: 0, interior: X1([Parachain(1004)]) }, what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(205217100000) }    
2024-05-30 21:13:44.723 TRACE main xcm::fungible_adapter: withdraw_asset what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(4687100000) }, who: Location { parents: 0, interior: X1([AccountId32 { network: None, id: [35, 154, 38, 60, 227, 140, 132, 230, 230, 19, 67, 42, 230, 243, 159, 250, 54, 60, 237, 56, 213, 43, 32, 204, 18, 211, 35, 95, 86, 125, 200, 237] }]) }    
2024-05-30 21:13:44.723 TRACE main xcm::fungible_adapter: can_check_out dest: Location { parents: 0, interior: X1([Parachain(1004)]) }, what: Asset { id: AssetId(Location { parents: 0, interior: Here }), fun: Fungible(4687100000) }
```

I didn't investigate it much further (just a little bit), but maybe the only difference I can see is `pallet_identity`'s deposits configuration. The benchmark `reap_identity` uses `pallet_identity::set_identity` which uses `calculate_identity_deposit` which uses `ByteDeposit` and `BasicDeposit`.

And the difference I see here:

For Polkadot: https://github.com/polkadot-fellows/runtimes/blob/main/relay/polkadot/src/lib.rs#L817-L820
```
        // Minimum 4 CENTS/byte
	pub const BasicDeposit: Balance = deposit(1, 258);
	pub const ByteDeposit: Balance = deposit(0, 1);
	pub const SubAccountDeposit: Balance = deposit(1, 53);
```

For Kusama: https://github.com/polkadot-fellows/runtimes/blob/main/relay/kusama/src/lib.rs#L974-L976
```
        // Minimum 100 bytes/KSM deposited (1 CENT/byte)
	pub const BasicDeposit: Balance = 1000 * CENTS;       // 258 bytes on-chain
	pub const ByteDeposit: Balance = deposit(0, 1);
	pub const SubAccountDeposit: Balance = 200 * CENTS;   // 53 bytes on-chain
```

Is it even correct `1000 * CENTS;       // 258 bytes on-chain` or `200 * CENTS;   // 53 bytes on-chain`?

I don't know maybe it is a false clue. But I suspect it could be related to the different ED on Polkadot vs Kusama.

Reply:
Yeah it is probably trying to put less than the ED into the `CheckAccount`, which doesn't exist in the test environment but does in production. I will fix.